cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs

```diff
@@ -996,11 +992,11 @@ getCurrentTimePlus30 =
 -- | Attempts to read Shelley genesis from disk
 -- and if not found creates a default Shelley genesis.
 readShelleyGenesisWithDefault
-  :: FilePath
+  :: File InOut
```

Comment:
This is an example of a file that reads from a file also writing to that same file.